comms/EclCC.cpp

```diff
@@ -725,11 +725,19 @@ boost::filesystem::path EclCCSubPath(const boost::filesystem::path & directory)
     return directory / "clienttools" / "bin" / "eclcc.exe";
 }
 
-unsigned int FindAllCEclCC(const boost::filesystem::path & progFiles, CEclCCVector & results)
+void PushUniqueEclCC(CEclCCVector& results, const boost::filesystem::path eclccPath)
 {
-    if (boost::filesystem::exists(progFiles)) {
+    bool found = false;
+    auto it = std::find_if(results.begin(), results.end(), [&eclccPath](const CEclCCAdapt& obj) {return (_tcsicmp(obj.get()->GetCompilerFilePath().c_str(), eclccPath.c_str()) == 0); });
```

Comment:
`std::any_of` returns a bool which may be preferable, but checking against `results.end()` is equally ok.